presto-docs/src/main/sphinx/security/internal-communication.rst

```diff
@@ -0,0 +1,85 @@
+=============================
+Secure Internal Communication
+=============================
+
+The Presto cluster can be configured to use secured communication. Communication
+between Presto nodes can be secured with encryption, authentication and authorization.
+
+.. Note: At this step only encryption is supported
+
+Encrypted Internal Communication Configuration
+----------------------------------------------
+
+Encryption is configured in `config.properties` file.
+Encryption properties are identical for all of the Presto node roles (coordinator, worker)
+and must be set on every node in the cluster.
+
+To enable encryption for the Presto internal communication follow the next steps:
+
+1. Disable HTTP endpoint.
+
+.. code-block:: none
+
+    http-server.http.enabled=false
+
+.. warning::
+
+    Leaving HTTP endpoint enabled leads to a security hole.
+
+2. Specify node's hosts. Every node must have it's own host configured.
+Hosts must be in the same parent domain to be able to easily generate custom SSL
+certificates. e.g.: `coordinator.example.com`, `worker1.example.com`, `worker2.example.com`.
+
+.. code-block:: none
+
+    node.external-address=<node-domain>
+
+3. Generate Java Keystore File. Every Presto node must be able to connect to any other node within
+the same cluster. It is possible to create private certificates for every single node for the
+exact domain, and than specify it for the server (`http-server.https.keystore.path`), and create a
+keystore that contains all the public keys for all the domains, and specify it for the
+client (`http-client.https.keystore.path`). But for simplicity single keystore for the base domain
```

Comment:
Suggest: "For simplicity, you can create a single keystore for the base domain with a wildcard"